RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -243,11 +621,195 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
         // make sure phi is in range -pi..pi
         if (sumPhiLayer[layer+startPositionToFill] > TMath::Pi())
           sumPhiLayer[layer+startPositionToFill] -= TMath::TwoPi();
-        newCluster.addToShapeParameters(sumEnLayer[layer+startPositionToFill] / E);
+
+        newCluster.addToShapeParameters(sumEnLayer[layer+startPositionToFill] / E);  // E fraction of layer
         newCluster.addToShapeParameters(sumThetaLayer[layer+startPositionToFill]);
         newCluster.addToShapeParameters(sumPhiLayer[layer+startPositionToFill]);
-      }
-    }
-  }
+
+        // do pi0/photon shape var only for EMB
+        if (m_do_pi0_photon_shapeVar && systemID == systemID_EMB) {
+          double w_theta = sqrt(fabs(theta2_E_layer[layer+startPositionToFill] / sumEnLayer[layer+startPositionToFill] - std::pow(theta_E_layer[layer+startPositionToFill] / sumEnLayer[layer+startPositionToFill], 2)));
+          if (std::isnan(w_theta))    w_theta = 0.;
```

Comment:
w_theta should be nan only if sumEnLayer of this layer is zero, shouldn't it? It would be better to replace this with something that handles this more cleanly and doesn't hide other NaN that might results from bugs in the code
```suggestion
          double w_theta = (sumEnLayer[layer+startPositionToFill] == 0.) ? 0. : sqrt(fabs(theta2_E_layer[layer+startPositionToFill] / sumEnLayer[layer+startPositionToFill] - std::pow(theta_E_layer[layer+startPositionToFill] / sumEnLayer[layer+startPositionToFill], 2)));
```